### PR TITLE
941 - Debug Batch implementation

### PR DESCRIPTION
**Pull request by avrohomgottlieb**
## Issue Number

Closes https://github.com/AlexsLemonade/scpca-portal/issues/941

## Purpose/Implementation Notes

This PR details the debugging changes made to the Batch implementation to produce workable infrastructure. 
Changes include:
- Adding a `batch.tf` module file to declare the batch subdirectory as a collection of related files
- Adding a `batch/variables.tf` file to enable resources and envars to make their way into the batch module
- Updating references and fixing typos to the current implementation

As well, this PR fixed all deprecations warnings by updating arguments and resources to their updated specifications.

## Types of changes

What types of changes does your code introduce?

- Bugfix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Functional tests

N/A

## Checklist

- [x] Lint and unit tests pass locally with my changes

## Screenshots

N/A

### CODE REVIEW

**Comment by avrohomgottlieb** [comment]
I'd like to propose a naming convention for resources in the `infrastructure/` directory.

Regarding the naming of resources, if there are multiple instances of a resource, then each instance should be given a name with a unique descriptor that distinguishes it some other instances of the same resource, such as `aws_cloudwatch_log_stream log_stream_api | log_stream_api_nginx_access | log_stream_api_nginx_error`. 

If there is only one instance of that resource, and we don't expect that there will be another, then it is recommended to name it according to the project, such as `aws_vpc scpca_portal`.

However, we should endeavor not to append the resource type to the end of the resource name, such as `aws_vpc scpca_portal_vpc`. The way that resources are referenced in terraform is by concatenating the resource type and the resource name with a dotted delimiter (dotted notation), so `aws_vpc.scpca_portal_vpc` is redundant.

A nuance with this convention should be with modules (like the `batch` module), where resources are only available if they are passed in as variables to the module. As such, the most straightforward way to name the variables should be to append the resource type to the resource name, which would look as follows.

```terraform
module "batch" {
    ...
    scpca_portal_vpc = aws_vpc.scpca_portal
}

**Comment by davidsmejia** [comment]
> I'd like to propose a naming convention for resources in the `infrastructure/` directory.
> 
> Regarding the naming of resources, if there are multiple instances of a resource, then each instance should be given a name with a unique descriptor that distinguishes it some other instances of the same resource, such as:
> 
> ```hcl
> resource "aws_subnet" "scpca_portal_1a" | "scpca_portal_1b"
> ```
> 
> If there is only one instance of that resource, and we don't expect that there will be another, then it is recommended to name it according to the project, such as:
> 
> ```hcl
> resource "aws_vpc" "scpca_portal"
> ```
> 
> However, we should endeavor not to append the resource type to the end of the resource name, such as `resource "aws_vpc" "scpca_portal_vpc"`. The way that resources are referenced in terraform is by concatenating the resource type and the resource name with a dotted delimiter (dotted notation), so `aws_vpc.scpca_portal_vpc` is redundant.
> 
> A nuance with this convention should be with modules (like the `batch` module), where resources are only available if they are passed in as variables to the module. As such, the most straightforward way to name the variables should be to append the resource type to the resource name, which would look as follows.
> 
> ```hcl
> module "batch" {
>     ...
>     scpca_portal_vpc = aws_vpc.scpca_portal
>     ...
> }
> ```

Thank you for suggesting this, could you remove this change from this PR and have it match what was previously defined. We should make that change in a separate effort.

File: infrastructure/batch/job_queue.tf [hunk 1]
```diff
@@ -3,11 +3,9 @@ resource "aws_batch_job_queue" "scpca_portal_project" {
...
   priority = 1
 
-  # compute_environments = [] array is deprecated
-  compute_environment_order {
-    order               = 1
-    compute_environment = aws_batch_compute_environments.scpca_portal_project.arn
-  }
+  compute_environments = [
+    aws_batch_compute_environment.scpca_portal_project.arn,
+  ]
```
Comment (avrohomgottlieb) [inline infrastructure/batch/job_queue.tf:8]: `compute_environments` is currently deprecated. The recommended approach is using `compute_environment_order`. Using `compute_environment_order` necessitates upgrading to aws provider version "5.40", which I ran into bugs while trying to implement. In the interest of time I kept `compute_environments` for now, but this should be updated in the future.

File: infrastructure/batch/roles.tf [hunk 2]
```diff
@@ -22,60 +23,59 @@ resource "aws_iam_role_policy_attachment" "batch_service_role" {
   policy_arn = "arn:aws:iam::aws:policy/service-role/AWSBatchServiceRole"
 }
 
-resource "aws_iam_role" "ecs_task_execution_role" {
+resource "aws_iam_role" "ecs_task_role" {
```
Comment (davidsmejia) [inline infrastructure/batch/roles.tf:26]: can you change these names back as well, or let me know if it is a required change

File: infrastructure/batch/job_definition.tf [hunk 3]
```diff
@@ -16,27 +16,27 @@ resource "aws_batch_job_definition" "scpca_portal_project" {
...
       },
       {
         name  = "DATABASE_HOST"
-        value = aws_db_instance.postgres_db.address
+        value = var.postgres_db.address
       },
       {
         name  = "DATABASE_PORT"
-        value = aws_db_instance.postgres_db.port
+        value = tostring(var.postgres_db.port)
```
[the remark this answers is not in the document]
Comment (avrohomgottlieb) [reply infrastructure/batch/job_definition.tf:31]: For some reason, `tostring()` adequately casts the port variable into a `str`, but the string interpolation does not succeed. Here is the error output from the string interpolation:

image

**Review by davidsmejia — APPROVED** [review]
LGTM

**Pull request merged**